src/app/contacts/new-contact/new-contact.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit, EventEmitter, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ContactsService } from '../contacts.service';
+
+@Component({
+  selector: 'app-new-contact',
+  templateUrl: './new-contact.component.html',
+  styleUrls: ['./new-contact.component.css']
+})
+
+export class NewContactComponent implements OnInit {
+ // @Output() addContact = new EventEmitter<any>();
+
+  isSaveButtonDisabled: boolean = true;
+  isSaveButtonVisible: boolean = false;
+
+  contactForm: FormGroup = this.fb.group({
+    firstName: ['', [Validators.required, Validators.minLength(3), Validators.pattern(/^\S*$/)]],
+    lastName: ['', [Validators.required, Validators.minLength(3), Validators.pattern(/^\S*$/)]],
+    email: ['', [Validators.required, Validators.email, Validators.pattern(/@boolean.co.uk$/)]],
+    phone: ['', [Validators.required, Validators.minLength(9)]]
+  });
+
+  constructor(private fb: FormBuilder, private contactService: ContactsService) { }
+
+  ngOnInit(): void {
+    this.initContactForm();
+  }
+
+  initContactForm() {
+    this.contactForm = this.fb.group({
```

Comment:
I think it would be a good idea to store the object outside of the class - then you could use it inside the class without repeating yourself.

Something like this

```ts
const InitialForm = {
    firstName: ['', [Validators.required, Validators.minLength(3), Validators.pattern(/^\S*$/)]],
    lastName: ['', [Validators.required, Validators.minLength(3), Validators.pattern(/^\S*$/)]],
    email: ['', [Validators.required, Validators.email, Validators.pattern(/@boolean.co.uk$/)]],
    phone: ['', [Validators.required, Validators.minLength(9)]]
  }
  
  
  export class NewContactComponent implements OnInit {
    // ...
    contactForm = this.gb.group(InitialForm)
    //...
    
    initContactForm() {
      this.contactForm = this.fb.group(InitialForm)
   }
  }
```